linkedlist/linkedList.go

```diff
@@ -108,3 +139,45 @@ func (l *LinkedList) HandleList() []int {
 	}
 	return values
 }
+
+func (l *LinkedList) SearchConcurrently(ctx context.Context, cancel context.CancelFunc, wg *sync.WaitGroup, find int) (int, bool) {
+	var result int
+	var isFound bool
+	var mu sync.Mutex
+
+	for i := 0; i < len(Nodes); i++ {
+		wg.Add(1)
+		go func(i int) {
+			defer wg.Done()
+			index := Nodes[i]
+			for j := 0; j < int(part); j++ {
+				select {
```

Comment:
As i know in this design, if one of this goroutines find the data it will call the cancel but others will continue because it only checks if the context is done at the start of the goroutine.
Please correct me if i am wrong